src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -110,86 +242,129 @@ dx_path() {
       ;;
   esac
 }
-    
+
 main() {
-    if [[ $debug == true ]]; then
-      export NXF_DEBUG=2
-      TRACE_CMD="-trace nextflow.plugin"
-      set -x && env | grep -v DX_SECURITY_CONTEXT | sort
-    fi
-    
-    if [ -n "$docker_creds" ]; then
-        dx-registry-login
-    fi
+  if [[ $debug == true ]]; then
+    export NXF_DEBUG=2
+    TRACE_CMD="-trace nextflow.plugin"
+    env | grep -v DX_SECURITY_CONTEXT | sort
+    set -x
+  fi
+
+  DX_CACHEDIR=$DX_PROJECT_CONTEXT_ID:/.nextflow_cache_db
+  validate_run_opts
+
+  if [[ $preserve_cache == true ]]; then
+    check_cache_db_storage
+  fi
+
+  if [ -n "$docker_creds" ]; then
+    dx-registry-login
+  fi
+
+  # set default NXF env constants
+  export NXF_DOCKER_LEGACY=true
+  #export NXF_DOCKER_CREDS_FILE=$docker_creds_file
+  #[[ $scm_file ]] && export NXF_SCM_FILE=$(dx_path $scm_file 'Nextflow CSM file')
+  export NXF_HOME=/opt/nextflow
+  export NXF_ANSI_LOG=false
+  export NXF_PLUGINS_DEFAULT=nextaur@1.1.0
+  export NXF_EXECUTOR='dnanexus'
+
+  # use /home/dnanexus/nextflow_playground as the temporary nextflow execution folder
+  mkdir -p /home/dnanexus/nextflow_playground
+  cd /home/dnanexus/nextflow_playground
+
+  # get job output destination
+  DX_JOB_OUTDIR=$(jq -r '[.project, .folder] | join(":")' /home/dnanexus/dnanexus-job.json)
+  # initiate log file
+  LOG_NAME="nextflow-$DX_JOB_ID.log"
+
+  # get current executable name
+  EXECUTABLE_NAME=$(jq -r .executableName /home/dnanexus/dnanexus-job.json)
+
+  # set/create current session id
+  if [[ -n $resume ]]; then
+    get_resume_session_id
+  else
+    NXF_UUID=$(uuidgen)
+  fi
+  export NXF_UUID
+  export NXF_CACHE_MODE=LENIENT
```

Comment:
If you could add a note above: "Using the lenient mode to caching makes it possible to reuse files for resume when taring/untaring cached files" or similar (so in the future we know the motivation